documentation/features/cpuset.md

```diff
@@ -0,0 +1,55 @@
+# Cpuset cgroup.
+
+From Kernel documentation:
+
+_" Cpusets provide a mechanism for assigning a set of CPUs and Memory Nodes to
+a set of tasks."_
+
+The Kata agent brings compatibility the cgroup cpuset CPU on the guest side.
+
+The cpuset CPU cgroup will be applied on two events:
+
+- containers creation.
+
+- container update.
+
+When the runtime request to apply cpuset cgroup to the agent, the amout of
+vcpus available may be not the same to the required in the request.
+
+This is becasue the request from the agent client (the Kata runtime) pass
+cpusets that were requested to be placed on the host. This is to isolate the
+container workload on some specific host CPUs. The runtime pass the
+requested cpuset to the agent, that will try to apply the cgroup cpuset on the
+guest.
+
+Because the runtime will only hot-plug and amount the CPUs calculated based in
+the container period and quota, the VM wont have the same amount CPUs than the host.
+guest.
+
+Example:
+
+docker run -ti --cpus 2 --cpuset 0,1 busybox
+ 
+The expected result is that the container is limited to the time of 2 CPUs, but
+is only allowed to be scheduled on CPUs 0 and 1 
+
+In a similar case and a valid traditional container is the following:
+
+docker run -ti --cpus 2 --cpuset 2,3,4 busybox
+
+Where the container is limited to 2 CPUs and can be scheduled on CPU 2,3,4.
+
+In the case of the Kata. The Kata runtime will only hotplug 2 cpus, making
+imposible to request to the guest kernel to schedule the workload on vcpu
+3,4.
+
+## cpuset best effort application.
+
+Kata will try agent will evaluate this request and see if is possible to 
```

Comment:
Lines 48-49 suggested rewrite (unsure if I've changed the intended meaning here): 
The Kata agent evaluates the request to see if it is possible to apply the cpuset request onto the guest.